src/core/lib/layer.js

```diff
@@ -449,6 +443,8 @@ export default class Layer {
     }
 
     this.clearChangeFlags();
+    // Release old props for GC once update is complete
+    this.oldProps = EMPTY_PROPS;
```

Comment:
If we do this, do we actually need to store them on the layer? If we keep them around and trigger another diff, we'll get incorrect results, so we might as well not store them at all? Is there any other reason to keep them around?

Reply:
`layerManager` assigns `oldProps` outside of the layer, which is layer used by `shouldUpdateState` and `updateState`. For minimum change we have to keep the field around.

Reply:
I agree that it should not be necessary.